From 366b03f607729c4538e91b634ecc57c8398522a1 Mon Sep 17 00:00:00 2001
From: Erwan Le Gall <legall.erwan@gmail.com>
Date: Tue, 26 Jan 2021 13:09:03 +0100
Subject: [PATCH] Update user.py

Double check on bind from user input in case of external access of the function
---
 passhportd/app/views_mod/user/user.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/passhportd/app/views_mod/user/user.py b/passhportd/app/views_mod/user/user.py
index f082da5f..20d7d1cd 100755
--- a/passhportd/app/views_mod/user/user.py
+++ b/passhportd/app/views_mod/user/user.py
@@ -28,7 +28,7 @@ def useruid(s, login):
 
     # Look for the user entry.
     if not c.search(config.LDAPBASE,
-                    "(" + config.LDAPFIELD + "=" + login + ")") :
+                    "(" + config.LDAPFIELD + "=" + escape_rdn(login) + ")") :
         app.logger.error("Error: Connection to the LDAP with service account failed")
     else:
         if len(c.entries) >= 1 :
@@ -49,7 +49,7 @@ def try_ldap_login(login, password):
     s = Server(config.LDAPURI, port=config.LDAPPORT,
                use_ssl=False, get_info=ALL)
     # 1. connection with service account to find the user uid
-    uid = useruid(s, login)
+    uid = useruid(s, escape_rdn(login))
    
     if uid: 
         # 2. Try to bind the user to the LDAP
